src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/DiscoverVariantsFromContigAlignmentsSAMSpark.java

```diff
@@ -174,52 +170,118 @@ public static AlignedContig parseReadsAndOptionallySplitGappedAlignments(final I
                 alignedContigs.filter(alignedContig -> alignedContig.alignmentIntervals.size()>1)                                     // filter out any contigs that has less than two alignment records
                         .mapToPair(alignedContig -> new Tuple2<>(alignedContig.contigSequence,                                        // filter a contig's alignment and massage into ordered collection of chimeric alignments
                                 ChimericAlignment.parseOneContig(alignedContig, DEFAULT_MIN_ALIGNMENT_LENGTH)))
                         .flatMapToPair(DiscoverVariantsFromContigAlignmentsSAMSpark::discoverNovelAdjacencyFromChimericAlignments)    // a filter-passing contig's alignments may or may not produce novel adjacency
-                        .groupByKey()                                                                                                 // group the same novel adjacency produced by different contigs together
+                        .groupByKey();                                                                                                // group the same novel adjacency produced by different contigs together
+
+        narlsAndSources.cache();
+
+        if ( parameters.truthVCF != null ) {
+            final SVIntervalTree<String> trueBreakpoints =
+                    SVVCFReader.readBreakpointsFromTruthVCF(parameters.truthVCF, sequenceDictionary, parameters.truthIntervalPadding);
+
+            if ( assembledIntervals != null ) {
+                evaluateIntervalsAgainstTruth(assembledIntervals, trueBreakpoints, localLogger);
+            }
+
+            final SVIntervalTree<String> narlyBreakpoints =
+                    readBreakpointsFromNARLs(narlsAndSources.map(Tuple2::_1).collect(), sequenceDictionary, parameters.truthIntervalPadding);
+
+            evaluateNARLsAgainstTruth(narlyBreakpoints, trueBreakpoints, localLogger);
+        }
+
+        List<VariantContext> annotatedVariants =
+                narlsAndSources
                         .mapToPair(noveltyAndEvidence -> inferType(noveltyAndEvidence._1, noveltyAndEvidence._2))                     // type inference based on novel adjacency and evidence alignments
                         .map(noveltyTypeAndEvidence ->
                                 annotateVariant(                                                                                      // annotate the novel adjacency and inferred type
                                         noveltyTypeAndEvidence._1,
                                         noveltyTypeAndEvidence._2._1,
                                         null,
                                         noveltyTypeAndEvidence._2._2,
-                                        broadcastReference));
+                                        broadcastReference))
+                        .collect();
 
-        List<VariantContext> collectedAnnotatedVariants = annotatedVariants.collect();
+        narlsAndSources.unpersist();
```

Comment:
Just since there was the Slack dust-up over .unpersist() vs .destroy(), I want to call to your attention to decide whether you wanted .unpersist() here. I'm still learning Spark and don't have a strong opinion, other than to note I've seen .destroy() crash in the wild, so I'd probably choose .unpersist() and accept the potential memory loss as you have here.

Reply:
That had to do with broadcasts.  This is an RDD.